modules/minifact/tasks/ruby.rb

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env ruby # rubocop:disable Lint/ScriptPermission
+# frozen_string_literal: true
+
+# Delegate to facter
+exec('facter', '--json')
```

Comment:
It'd probably make sense to bring in all of https://github.com/puppetlabs/puppetlabs-facter_task/blob/master/tasks/init.rb, which has already dealt with some issues around finding Facter.

Reply:
Oh, the implementation is different because it's only returning a single fact.

Reply:
Yeah, I don't know what's going on with that module then. Maybe no one's used it with structured facts?

Reply:
Yeah, it works but doesn't return anything readable. Comes back as
```
{
  "status" : "{\n  architecture => \"x86_64\",\n  family => \"RedHat\",\n  hardware => \"x86_64\",\n  name => \"CentOS\",\n  release => {\n    full => \"7.2.1511\",\n    major => \"7\",\n    minor => \"2\"\n  },\n  selinux => {\n    config_mode => \"enforcing\",\n    config_policy => \"targeted\",\n    current_mode => \"enforcing\",\n    enabled => true,\n    enforced => true,\n    policy_version => \"28\"\n  }\n}"
}
```

Reply:
My comment was unsubstantiated, the JSON parse is parsing the task's input parameters rather than the facter output, as I was thinking when I wrote the comment.
But still the value currently returned for structured facts is not very useful for machine processing.